### ISSUE TITLE
FocusManager skipFocus issue

### ISSUE THREAD

**Issue by CooperCao** [rdkcentral/Lightning-UI-Components#37]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `@lightningjs/ui-components@1.0.0` for the project I'm working on.

If skipfocus is set for the first or last item, the focus will be incorrectly switched.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/@lightningjs/ui-components/components/FocusManager/index.js b/node_modules/@lightningjs/ui-components/components/FocusManager/index.js
index 587f1aa..0799287 100644
--- a/node_modules/@lightningjs/ui-components/components/FocusManager/index.js
+++ b/node_modules/@lightningjs/ui-components/components/FocusManager/index.js
@@ -91,10 +91,11 @@ export default class FocusManager extends lng.Component {
   }
 
   // Override
-  render() {}
+  render() { }
 
   selectPrevious() {
     if (this.selectedIndex > 0) {
+      let currentIndex = this.selectedIndex;
       let prevIndex = this.selectedIndex - 1;
       let previous = this.items[prevIndex];
       while (prevIndex && previous.skipFocus) {
@@ -103,7 +104,10 @@ export default class FocusManager extends lng.Component {
         prevIndex -= 1;
         previous = this.items[prevIndex];
       }
-      this.selectedIndex = prevIndex;
+       if (previous.skipFocus && (prevIndex == 0))
+        this.selectedIndex = currentIndex;
+      else
+        this.selectedIndex = prevIndex;
       return true;
     } else if (this.wrapSelected) {
       this.selectedIndex = this.Items.children.length - 1;
@@ -114,6 +118,7 @@ export default class FocusManager extends lng.Component {
 
   selectNext() {
     if (this.selectedIndex < this.Items.children.length - 1) {
+      let currentIndex = this.selectedIndex;
       let nextIndex = this.selectedIndex + 1;
       let next = this.items[nextIndex];
       while (nextIndex < this.items.length - 1 && next.skipFocus) {
@@ -122,7 +127,10 @@ export default class FocusManager extends lng.Component {
         nextIndex += 1;
         next = this.items[nextIndex];
       }
-      this.selectedIndex = nextIndex;
+      if (next.skipFocus && (nextIndex == this.items.length - 1))
+        this.selectedIndex = currentIndex;
+      else
+        this.selectedIndex = nextIndex;
       return true;
     } else if (this.wrapSelected) {
       this.selectedIndex = 0;
```

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by CooperCao** [comment]
For more information:
if skipfocus is set for the first item, the focus entered for the first time is not correct. I think I need to find the nearest item in the selectedIndex.

**Comment by erautenberg** [comment]
@ZhipengJuly thank you for your feedback and I apologize for the long delay. We are currently working to overhaul this repo, so a new version of the FocusManager should be available in the next week or so that addresses this exact concern.

**Comment by anthony9187** [comment]
Hi @ZhipengJuly, this has been resolved in the latest release (v1.1.0). I'm going to close this issue now, thanks again for your feedback!
